File: 102. Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        q = []
        q.append((root, 0))
        m = {}
        while q:
            node, lvl = q.pop(0)
            if not node:
                continue
            if lvl in m:
                m[lvl].append(node.val)
            else:
                m[lvl] = [node.val]
                
            q.append((node.left, lvl+1))
            q.append((node.right, lvl+1))
        
        print(m)
        rs = []
        for ind, ls in m.items():
            rs.append(ls)
            
        return rs

File: cpp/102_Binary_Tree_Level_Order_Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void level_order_traversal(TreeNode* root,vector<vector<int>>&ans,int height)
    {
        if(root == NULL)
        {
            return;
        }
        if(height > ans.size())
        {
            vector<int>var;
            var.push_back(root->val);
            ans.push_back(var);
        }
        else
        {
            ans[height-1].push_back(root->val);
        }
        level_order_traversal(root->left,ans,height+1);
        level_order_traversal(root->right,ans,height+1);
    }
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>>ans;
        int height = 1;
        level_order_traversal(root,ans,height);
        return ans;
    }
};